eng/mark-shipped.ps1

```diff
@@ -0,0 +1,51 @@
+[CmdletBinding(PositionalBinding=$false)]
+param ()
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+function MarkShipped([string]$dir) {
+    $shippedFilePath = Join-Path $dir "PublicAPI.Shipped.txt"
+    $shipped = Get-Content $shippedFilePath
+    if ($null -eq $shipped) {
+        $shipped = @()
+    }
+
+    $unshippedFilePath = Join-Path $dir "PublicAPI.Unshipped.txt"
+    $unshipped = Get-Content $unshippedFilePath
+    $removed = @()
+    $removedPrefix = "*REMOVED*";
+    Write-Host "Processing $dir"
+
+    foreach ($item in $unshipped) {
+        if ($item.Length -gt 0) {
+            if ($item.StartsWith($removedPrefix)) {
+                $item = $item.Substring($removedPrefix.Length)
+                $removed += $item
+            }
+            elseif ($item -ne "#nullable enable") {
+                $shipped += $item
+            }
+        }
+    }
+
+    $shipped | Sort-Object | Where-Object { -not $removed.Contains($_) } | Out-File $shippedFilePath -Encoding Ascii
+    "" | Out-File $unshippedFilePath -Encoding Ascii
+}
+
+try {
+    Push-Location "$PSScriptRoot\..\src"
+
+    foreach ($file in Get-ChildItem -re -in "PublicApi.Shipped.txt") {
+        $dir = Split-Path -parent $file
+        MarkShipped $dir
+    }
+}
+catch {
+    Write-Host $_
+    Write-Host $_.Exception
+    exit 1
```

Comment:
this is something to avoid, you will see it work just fine here, but in other scripts that are not so standalone, this will exit the parent script when you dot-source this script to get the functions defined in it, leading to super confusing errors that are hard to track.